install/kubernetes/cilium/templates/clustermesh-apiserver/service.yaml

```
@@ -24,4 +24,6 @@ spec:
   {{- if and (eq "LoadBalancer" .Values.clustermesh.apiserver.service.type) .Values.clustermesh.apiserver.service.loadBalancerIP }}
   loadBalancerIP: {{ .Values.clustermesh.apiserver.service.loadBalancerIP }}
   {{- end }}
+  externalTrafficPolicy: {{ .Values.clustermesh.apiserver.service.externalTrafficPolicy }}
+  internalTrafficPolicy: {{ .Values.clustermesh.apiserver.service.internalTrafficPolicy }}
```

This field (`internalTrafficPolicy`) is only available in Kubernetes 1.26, which s very recent and thus this manifest will fail on older installations.

Let's therefore make it optional. I would suggest the following:

- Set the `.Values.clustermesh.apiserver.service.externalTrafficPolicy` and `.Values.clustermesh.apiserver.service.internalTrafficPolicy` to empty strings by default (in the values.yml.tmpl file)
- Gate those two fields here via an if statement as follows:


```suggestion
  {{- if .Values.clustermesh.apiserver.service.externalTrafficPolicy }}
  externalTrafficPolicy: {{ .Values.clustermesh.apiserver.service.externalTrafficPolicy }}
  {{- end }}
  {{- if .Values.clustermesh.apiserver.service.internalTrafficPolicy }}
  internalTrafficPolicy: {{ .Values.clustermesh.apiserver.service.internalTrafficPolicy }}
  {{- end }}
```

---

@gandro 
Thx!
Fixed and squashed to one commit.
How do I add your contribution to the commit?